src/main/java/ch/njol/skript/effects/EffLook.java

```diff
@@ -106,12 +106,14 @@ protected void execute(Event event) {
 		Object object = target.getSingle(event);
 		if (object == null)
 			return;
-		Float speed = this.speed == null ? null : this.speed.getSingle(event).floatValue();
-		Float maxPitch = this.maxPitch == null ? null : this.maxPitch.getSingle(event).floatValue();
+
+		Number speed = this.speed == null ? null : this.speed.getOptionalSingle(event).orElse(null);
+		Number maxPitch = this.maxPitch == null ? null : this.maxPitch.getOptionalSingle(event).orElse(null);
+
 		if (LOOK_ANCHORS) {
-			PaperEntityUtils.lookAt(anchor, object, speed, maxPitch, entities.getArray(event));
+			PaperEntityUtils.lookAt(anchor, object, (Float) speed, (Float) maxPitch, entities.getArray(event));
```

Comment:
Actually, I don't believe it is safe to cast to here. You need to use .floatValue() (`speed` or `maxPitch` could be an Integer)

Reply:
piggybacking to say add a regression test too